plugins/speculation-rules/helper.php

```diff
@@ -29,21 +29,29 @@ function plsr_get_speculation_rules(): array {
 	$prefixer = new PLSR_URL_Pattern_Prefixer();
 
 	$base_href_exclude_paths = array(
-		$prefixer->prefix_path_pattern( '/wp-login.php', 'site' ),
+		$prefixer->prefix_path_pattern( '/wp-*.php', 'site' ),
 		$prefixer->prefix_path_pattern( '/wp-admin/*', 'site' ),
-		$prefixer->prefix_path_pattern( '/*\\?*(^|&)_wpnonce=*', 'home' ),
 		$prefixer->prefix_path_pattern( '/*', 'uploads' ),
 		$prefixer->prefix_path_pattern( '/*', 'content' ),
 		$prefixer->prefix_path_pattern( '/*', 'plugins' ),
 		$prefixer->prefix_path_pattern( '/*', 'template' ),
 		$prefixer->prefix_path_pattern( '/*', 'stylesheet' ),
 	);
 
+	/*
+	 * If pretty permalinks are enabled, exclude any URLs with query parameters.
+	 * Otherwise, exclude specifically the URLs with a `_wpnonce` query parameter.
+	 */
+	if ( (bool) get_option( 'permalink_structure' ) ) {
+		$base_href_exclude_paths[] = $prefixer->prefix_path_pattern( '/*\\?(.+)', 'home' );
```

Comment:
Nit: No need for the match group:

```suggestion
		$base_href_exclude_paths[] = $prefixer->prefix_path_pattern( '/*\\?.+', 'home' );
```

Reply:
Did you test this? I got an error when I tested this in the browser without the parentheses.

Reply:
Oops. You're right.

👎🏻 https://urlpattern.com/#eyJ1cmxQYXR0ZXJuIjoiaHR0cHM6Ly93b3JkcHJlc3Mub3JnL1xcPy4rIiwidXJsIjoiaHR0cHM6Ly93b3JkcHJlc3Mub3JnLz9mb28ifQ==

👍🏻 
https://urlpattern.com/#eyJ1cmxQYXR0ZXJuIjoiaHR0cHM6Ly93b3JkcHJlc3Mub3JnL1xcPyguKykiLCJ1cmwiOiJodHRwczovL3dvcmRwcmVzcy5vcmcvP2ZvbyJ9